src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/job/WorkerQueue.java

```diff
@@ -0,0 +1,54 @@
+package de.dhbw.karlsruhe.students.mailflow.core.application.job;
+
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+public class WorkerQueue {
+  private static WorkerQueue INSTANCE;
+
+  private PriorityQueue<Job> queue;
+
+  private WorkerQueue() {
+    this.queue = new PriorityQueue<>(new Comparator<Job>() {
+      @Override
+      public int compare(Job o1, Job o2) {
+        return o1.getReadyTime().compareTo(o2.getReadyTime());
+      }
+    });
+  }
+
+  public static WorkerQueue getInstance() {
```

Comment:
We learned that singletons can and should be implemented as enum. Would be interesting to see it in this project. This is also fine for me. It was just a reminder :D

Reply:
according to stackoverflow, this is completely up to personal taste.
I think `WorkerQueue` is big enough it deserves to be labeled as a class :D